### DISCUSSION TITLE
Building apps with multiple webviews

### DISCUSSION THREAD

**Discussion by kishaningithub** [wailsapp/wails#1163] [Q&A]
What does it take to build something like https://getferdi.com/ in wails?

Are webviews(whatsapp, gmail etc) within webview(parent application) supported?

Alternate ways to build that in wails without using multiple webviews are also welcome

PS - I am very new to Wails.

**Comment by leaanthony** [comment]
Hi and welcome! The project doesn't currently support multiple windows (it's unclear if the app you linked to does) but it is on the cards to support them for v3. Regarding webviews in webviews, I think the only way to do that is using the latest IFrame alternative (is there one?).

Edit: looks like it proxies 3rd party sites via a proxy server. Can't quite see how that's injected into the main view but it should be possible to recreate this.

**Reply by kishaningithub** [reply]
Mainly it uses browser view of electron

https://www.electronjs.org/docs/latest/api/browser-view

to embed multiple websites and also same websites with different credentials.

Its a stand alone desktop app, you can completely ignore the server sides.

Another similar alternative to the above is rambox

https://github.com/ramboxapp/community-edition/blob/master/README.md

If you see the privacy part of the above it says

Sessions will persist using the [partition:persist](https://electronjs.org/docs/api/webview-tag#partition) attribute for Webviews.
So every time you open Rambox, your sessions will keep alive until you remove the service.

Also iframes are not fine for this usecase because the credentials/session have to be sandboxed to support same website different credentials scenario and so for security reasons.

Both of these though functionally fine hogs a lot of system resources mainly because of electron IMO, i was thinking of creating an alternative which is functionally equivalent but with minimal resource hog.

**Reply by kishaningithub** [reply]
> it should be possible to recreate this.

@leaanthony When you said this, what was the approach(es) you had in mind?
